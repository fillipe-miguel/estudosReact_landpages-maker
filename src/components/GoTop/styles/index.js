import styled, { css } from 'styled-components';

export const GoTopRoot = styled.a`
  ${({ theme }) => css`
    align-items: center;
    background-color: ${theme.colors.primaryColor};
    border-radius: 1rem;
    border: 2px solid ${theme.colors.white};
    bottom: 2rem;
    color: ${theme.colors.white};
    display: flex;
    font-size: ${theme.fonts.sizes.medium};
    height: 4rem;
    justify-content: center;
    position: fixed;
    right: 2rem;
    opacity: 0.5;
    z-index: 6;
    transition: all 0.3s ease-in-out;
    width: 4rem;

    &:hover {
      opacity: 1;
      border-color: ${theme.colors.secondaryColor};
      box-shadow: 0px 0px 8px 1px ${theme.colors.secondaryColor};
      color: ${theme.colors.secondaryColor};
      transform: scale(1.05);
      transition: all 0.3s ease-in-out;
    }

    &:hover > :first-child {
      animation: shakeUpDown 0.7s ease-in-out infinite;
    }
  `}

  @keyframes shakeUpDown {
    from {
      transform: translateY(0px);
    }
    to {
      transform: translateY(-5px);
    }
  }
`;

export default GoTopRoot;
