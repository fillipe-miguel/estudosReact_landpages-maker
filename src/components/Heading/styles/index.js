import styled, { css } from 'styled-components';

export const HeadingRoot = styled.h1`
  // ================================= Para pegar as props =================================
  ${({ theme, ...props }) => css`
    color: ${props.colorDarker
      ? theme.colors.primaryColor
      : theme.colors.white};

    text-transform: ${props.uppercase ? 'uppercase' : 'none'};

    font-size: ${theme.fonts.sizes[props.size]};
  `}

  // ================================= Medias =================================
  ${({ theme, ...props }) => css`
    @media ${theme.medias.lteMedium} {
      font-size: ${props.size === 'xxxlarge'
        ? theme.fonts.sizes.xlarge
        : theme.fonts.sizes[props.size]};
    }
  `}
`;

export default HeadingRoot;

// ================================= Antiga organização =================================

// font-size: ${({ theme, size }) => titleSize[size](theme)};

// @media ${({ theme }) => theme.media.lteMedium} {
//   font-size: ${({ theme, size }) =>
//     size === 'xxxlarge'
//       ? theme.fonts.sizes.xlarge
//       : theme.fonts.sizes.xxxlarge};
// }

// color: ${({ theme, colorDarker }) =>
//     colorDarker ? theme.colors.black : theme.colors.white};

// text-transform: ${({ uppercase }) => (uppercase ? 'uppercase' : 'none')};
// font-size: ${({ theme, size }) => theme.fonts.sizes[size]};
