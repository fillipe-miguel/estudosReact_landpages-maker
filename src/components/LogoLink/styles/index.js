import styled, { css } from 'styled-components';

export const LogoLinkRoot = styled.a`
  ${() => css`
    display: flex;
    align-items: center;

    > img {
      max-height: 3rem;
    }
  `}
`;

export default LogoLinkRoot;
