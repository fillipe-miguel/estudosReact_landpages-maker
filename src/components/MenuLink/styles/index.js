import styled, { css } from 'styled-components';

export const MenuLinkRoot = styled.a`
  ${({ theme }) => css`
    display: inline-block;
    text-decoration: none;
    position: relative;
    font-size: ${theme.fonts.sizes.medium};
    padding: ${theme.spacings.small};
    color: ${theme.colors.primaryColor};

    &::after {
      content: '';
      position: absolute;
      bottom: 1rem;
      left: 50%;
      width: 0;
      height: 0.2rem;
      background-color: ${theme.colors.secondaryColor};
      transition: all 300ms ease-in-out;
    }

    &:hover::after {
      left: 25%;
      width: 50%;
    }
  `}
`;

export default MenuLinkRoot;
