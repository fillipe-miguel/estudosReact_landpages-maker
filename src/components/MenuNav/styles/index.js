import styled, { css } from 'styled-components';

export const MenuNavRoot = styled.nav`
  ${({ theme }) => css`
    display: flex;
    flex-flow: row wrap;

    @media ${theme.medias.lteMedium} {
      flex-flow: column nowrap;
      align-items: center;
    }
  `}
`;

export default MenuNavRoot;
