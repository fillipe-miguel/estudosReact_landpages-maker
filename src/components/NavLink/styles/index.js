import styled, { css } from 'styled-components';
import medias from './medias';

export const NavLinkRoot = styled.nav`
  ${({ theme }) => css`
    ${medias(theme)}
    display: flex;
    flex-flow: row wrap;
  `}
`;

export default NavLinkRoot;
