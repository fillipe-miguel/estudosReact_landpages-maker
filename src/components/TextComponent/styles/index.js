import styled, { css } from 'styled-components';

export const TextComponentRoot = styled.p`
  ${({ theme, ...props }) => css`
    font-size: ${theme.fonts.sizes.small};
    color: ${props.colorDarker
      ? theme.colors.primaryColor
      : theme.colors.white};
  `}
`;

export default TextComponentRoot;
