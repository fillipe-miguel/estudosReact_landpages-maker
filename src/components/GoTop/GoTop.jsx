import React from 'react';

import { MdArrowUpward } from 'react-icons/md';

import * as Styled from './styles';

function GoTop() {
  return (
    <Styled.GoTopRoot
      href="#"
      aria-label="Button go to top"
      title="Voltar ao topo"
    >
      <MdArrowUpward />
    </Styled.GoTopRoot>
  );
}

export default GoTop;
