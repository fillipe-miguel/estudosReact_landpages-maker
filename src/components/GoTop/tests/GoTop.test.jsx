/* eslint-disable no-undef */
import { render, screen } from '@testing-library/react';

import GoTop from '..';

import { MyThemeProvider } from '../../../styles';

describe('<GoTop />', () => {
  // ================================= test 1 =================================
  it('Should render default', () => {
    render(
      <MyThemeProvider>
        <GoTop />
      </MyThemeProvider>,
    );

    expect(
      screen.getByRole('link', { name: 'Button go to top' }),
    ).toHaveAttribute('href', '#');
  });

  // ================================= Snapshot =================================
  it('Should render default', () => {
    const { container } = render(
      <MyThemeProvider>
        <GoTop />
      </MyThemeProvider>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
