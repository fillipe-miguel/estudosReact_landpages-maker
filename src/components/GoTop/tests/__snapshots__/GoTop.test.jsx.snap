// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GoTop /> Should render default 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #0A1128;
  border-radius: 1rem;
  border: 2px solid #FFF;
  bottom: 5rem;
  color: #FFF;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 2.4rem;
  height: 4rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: fixed;
  right: 2rem;
  opacity: 0.5;
  z-index: 6;
  -webkit-transition: all 0.3s ease-in-out;
  transition: all 0.3s ease-in-out;
  width: 4rem;
}

.c0:hover {
  opacity: 1;
  border-color: #DC143C;
  box-shadow: 0px 0px 8px 1px #DC143C;
  color: #DC143C;
  -webkit-transform: scale(1.05);
  -ms-transform: scale(1.05);
  transform: scale(1.05);
  -webkit-transition: all 0.3s ease-in-out;
  transition: all 0.3s ease-in-out;
}

.c0:hover > :first-child {
  -webkit-animation: shakeUpDown 0.7s ease-in-out infinite;
  animation: shakeUpDown 0.7s ease-in-out infinite;
}

<a
  aria-label="Button go to top"
  class="c0"
  href="#"
  title="Voltar ao topo"
>
  <svg
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    stroke-width="0"
    viewBox="0 0 24 24"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M0 0h24v24H0V0z"
      fill="none"
    />
    <path
      d="M4 12l1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
    />
  </svg>
</a>
`;
